price_service/server-rust/src/main.rs

```diff
@@ -0,0 +1,128 @@
+#![feature(never_type)]
+
+use {
+    anyhow::Result,
+    futures::{
+        channel::mpsc::Receiver,
+        SinkExt,
+    },
+    std::time::Duration,
+    structopt::StructOpt,
+    tokio::{
+        spawn,
+        time::sleep,
+    },
+};
+
+mod config;
+mod network;
+
+/// A Wormhole VAA is an array of bytes. TODO: Decoding.
+#[derive(Debug, Clone, Eq, Hash, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct Vaa {
+    pub data: Vec<u8>,
+}
+
+/// A PythNet AccountUpdate is a 32-byte address and a variable length data field.
+///
+/// This type is emitted by the Geyser plugin when an observed account is updated and is forwrarded
+/// to this process via IPC.
+#[derive(Debug, Clone, Eq, Hash, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct AccountUpdate {
+    addr: [u8; 32],
+    data: Vec<u8>,
+}
+
+/// Handler for LibP2P messages. Currently these consist only of Wormhole Observations.
+fn handle_message(_observation: network::p2p::Observation) -> Result<()> {
+    println!("Rust: Received Observation");
+    Ok(())
+}
+
+/// Initialize the Application. This can be invoked either by real main, or by the Geyser plugin.
+async fn init(_update_channel: Receiver<AccountUpdate>) -> Result<()> {
+    log::info!("Initializing PythNet...");
```

Comment:
price service -- couple more instances of "PythNet" in comments here, suggest find/replace.

Reply:
Ditto to above comment, will clarify soon.